src/pal/src/debug/debug.cpp

```diff
@@ -76,6 +76,10 @@ extern "C" void DBG_DebugBreak_End();
 #define CTL_WAIT        "wait"
 #endif   // HAVE_PROCFS_CTL
 
+#ifndef PROCFS_MEM_NAME
+#define PROCFS_MEM_NAME "mem"
```

Comment:
I would prefer not requiring /proc file system to be enabled on FreeBSD. I have thought that when the FreeBSD support was initially brought up, we didn't require /proc file system to be enabled, but I may be wrong.
So if this is the only /proc fs requirement in CoreCLR, could we implement accessing foreign process memory differently? We have a code that uses ptrace as an alternative in the sources already, so it might just work.
It seems that we could change the code that detects HAVE_PROCFS_CTL in configure.cmake so that it reports false on FreeBSD even if you have /proc fs enabled at the build time. That would make sure that everything would work even for people that don't have /proc fs enabled.